### DISCUSSION TITLE
My project wont deploy if i have vercel.json file.

### DISCUSSION THREAD

**Discussion by GitVex** [vercel/community#2315] [Help] [answered]
### Summary

I recently wanted to add cron jobs to my vercel project through adding a vercel.json file in my repository. Although, I can't seem to deploy the branch into preview. 

vercel.json
`
{
    "crons": [
        {
            "path": "/api/cron",
            "schedule": "* * * * *"
        }
    ]
}
`

I looked through the documentation and couldnt find anything that would imply that the file needs some sort of boilerplate to function, so I am just assuming that it adds to the project configuration.

### Additional information

I have already tried to investiagte solutions after reading a similar discussion (https://github.com/orgs/vercel/discussions/1500) but I don't seem to have a problem with an invisible character in the file. There is another user that appears to have had a problem similar to mine, though I couldn't find any threads made by them that refers to the topic. 

### Example

Access to the repository is public: https://github.com/GitVex/BardicInspiration

**Comment by amyegan** [comment]
Hi @GitVex! Thanks for sharing the repo. Which branch should I use to see your non-functioning vercel.json in context? If you can share the error details, that would also be helpful 🙏

**Reply by GitVex** [reply]
Hey! Sorry, that was a bit of an oversight. The related branch is `20-feature-remove-unavailable-links` https://github.com/GitVex/BardicInspiration/tree/20-feature-remove-unavailable-links

**Reply by GitVex** [reply]
Odd. On the github branch view it displays that the preview deployment has failed. Though on the project deployments tab on vercel.com it shows that it built successfully.

Also, it kind of dawns on me right now that cron jobs might only be run on production deployments?

**Reply by amyegan** [reply, marked as answer]
🙂 Thanks for sharing the branch. That vercel.json config looks correct.

You're right that cron jobs only run on production deployments. Another thing that sometimes catches people is that Hobby accounts are limited to 2 cron jobs that can be triggered once per day, but the limits are much higher for pro accounts: [Cron Job Limits](https://vercel.com/docs/cron-jobs#limits)

**Reply by GitVex** [reply]
Ah good! Thank you for the fast answer. I should set the job schedule back up to once per day then, maybe that will fix the failed deployment icon.

**Comment by foolo** [comment]
> 🙂 Thanks for sharing the branch. That vercel.json config looks correct.
> 
> You're right that cron jobs only run on production deployments. Another thing that sometimes catches people is that Hobby accounts are limited to 2 cron jobs that can be triggered once per day, but the limits are much higher for pro accounts: [Cron Job Limits](https://vercel.com/docs/cron-jobs#limits)

@amyegan Hi, I'm borrowing this thread since I have the exact same problem: The deployment fails without logs when I add a vercel.json with a cronjob. On the commit in GitHub, there's a red X, but the  "Details" link on the error dialog can't be clicked. See screen recording below.

I assume that there's a problem with my vercel.json file.

My question is:
**If f there is an error in the vercel.json, where do I find the error messages?**


https://github.com/vercel/community/assets/667227/ba9696ee-7664-459f-8f88-db2638c8ad6a

The commit in question: https://github.com/foolo/dkim-lookup/commit/1d94f849e4e0f2e753bc01ac85fcb2d373d9ab0a

**Reply by amyegan** [reply]
Hi @foolo. I can't see the video you posted. But the GitHub commit will usually have a comment with error info if the problem is related to `vercel.json` config. 

Something I find helpful when debugging build issues using [Vercel CLI](https://vercel.com/docs/cli) from my local computer. That will log error information to the terminal. If you run [`vercel build`](https://vercel.com/docs/cli/build#) on your local repo, do you see more details about the error?

**Reply by foolo** [reply]
@amyegan   Hey, thanks for replying so fast!

Here's the video on Youtube instead: https://youtu.be/ccHT0501Xz0

You wrote:
> But the GitHub commit will usually have a comment with error info if the problem is related to vercel.json config.


Although [this page](https://vercel.com/guides/why-aren-t-commits-triggering-deployments-on-vercel#look-for-commit-comments)  seems to say that there will be no such comments if vercel.json is incorrect:

> If the Vercel for Git integration is installed properly, the Vercel Bot will leave comments on your commits that indicate success or failure. **These comments may not always be present** on the pull request if no Deployment could be created, this occurs when pre-deployment validation fails, **for example, an incorrectly formatted vercel.json file.**

I did as you suggested tried with `vercel build` on the failing commit, but it works just fine locally. It seems to fail only on Github.

Although, for this particular issue, I have now verified that the problem was that my cronjob was too frequent for Hobby plan. If I change from `10 * * * *` to `10 17 * * *`, the deployment works fine. But it was pure luck that I found this thread and connected the dots. So it would still be handy if it were possible to get see error messages when deployment failed because of vercel.json.
